### PR TITLE
TaskList Component DG

### LINKED ISSUE #111: Developer Guide

### PR DESCRIPTION
**Pull request by wish2023**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -29,9 +29,14 @@ The rest of the App consists of these components.
 
 ### TaskList component
 
+![TaskListStructure](https://raw.githubusercontent.com/wish2023/tp/DG-tasklist/docs/structures/TaskListStructure.png)
+
 **API** :
 [`TaskList.java`](https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/master/src/main/java/athena/TaskList.java)
 
+1. The `TaskList` stores task data in `Task` type objects.
+2. The `TaskList` is updated in `Athena`.
```
Comment (daniellimws) [inline docs/DeveloperGuide.md:38]: After #131 it will be in `LogicManager`.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -29,9 +29,14 @@ The rest of the App consists of these components.
...
+3. `Task` stores
```
Comment (daniellimws) [inline docs/DeveloperGuide.md]: Lol you forgot to finish? Or we continue this next time?

File: src/main/java/athena/task/Time.java [hunk 3]
```diff
@@ -38,6 +38,7 @@ public Time(Boolean isFlexible, String startTime, String duration, String deadli
         this.recurrence = recurrence;
 
         setRecurrence(recurrence);
+        assert recurrenceDates.size() != 0;
```
Comment (daniellimws) [inline src/main/java/athena/task/Time.java]: Strictly speaking this shouldn't be part of this PR. But this is good.

**Review by daniellimws — APPROVED** [review]
Left some comments but I think this is ok to merge since we might make more changes later.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wish2023) [reply docs/DeveloperGuide.md:38]: Yup let's change it to `logicManager` once that's merged

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (wish2023) [reply docs/DeveloperGuide.md]: Oh whoops

File: src/main/java/athena/task/Time.java (hunk 3 above)
Comment (wish2023) [reply src/main/java/athena/task/Time.java]: Okay I'll delete this and add it in another branch
